AGENTS.md

```diff
@@ -0,0 +1,271 @@
+npm test
+```
+
+### Linting
+
+**Go linting:**
+```bash
+golangci-lint run
+```
+
+**Frontend linting:**
+```bash
+cd web_ui/frontend
+npm run lint
+```
+
+**Frontend formatting:**
+```bash
+cd web_ui/frontend
+npm run format        # Check formatting
+npm run format:fix    # Fix formatting
+```
+
+## Code Conventions
+
+### Go Code
+
+1. **License Headers**: All Go files must include the Apache 2.0 license header at the top (use current year):
+   ```go
+   /***************************************************************
+    *
+    * Copyright (C) {YEAR}, Pelican Project, Morgridge Institute for Research
+    *
+    * Licensed under the Apache License, Version 2.0 (the "License"); you
+    * may not use this file except in compliance with the License.  You may
+    * obtain a copy of the License at
+    *
+    *    http://www.apache.org/licenses/LICENSE-2.0
+    *
+    * Unless required by applicable law or agreed to in writing, software
+    * distributed under the License is distributed on an "AS IS" BASIS,
+    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    * See the License for the specific language governing permissions and
+    * limitations under the License.
+    *
+    ***************************************************************/
+   ```
+
+2. **Package Naming**: Use lowercase package names matching the directory name
+
+3. **Error Handling**: Always check and handle errors appropriately
+
+4. **Testing**: Place test files alongside source files with `_test.go` suffix
+
+5. **Imports**: Use `goimports` for automatic import organization
+
+### TypeScript/React Code
+
+1. **Components**: Use functional components with hooks
+
+2. **Type Safety**: Prefer explicit types over `any`
+
+3. **File Organization**: 
+   - Components in `web_ui/frontend/components/`
+   - Pages/routes in `web_ui/frontend/app/`
+   - Utilities in `web_ui/frontend/helpers/`
+   - Custom hooks in `web_ui/frontend/hooks/`
+
+4. **Styling**: Use Material-UI components and styling system
+
+5. **Formatting**: Follow Prettier configuration (`.prettierrc.json`)
+
+## Development Workflow
+
+### Environment Setup
+
+Pelican uses Docker containers and VSCode Dev Containers for development. See [CONTRIBUTE.md](CONTRIBUTE.md) for detailed setup instructions.
+
+Key steps:
+1. Fork and clone the repository
+2. Install Docker and VSCode
+3. Pull the development container: `hub.opensciencegrid.org/pelican_platform/pelican-dev:latest-itb`
+4. Create `.devcontainer/devcontainer.json` using the template from `dev/devcontainer.json`
+5. Build Pelican from source inside the container
+
+### Pull Request Guidelines
+
+- Address a single concern with minimal changes
+- Add tests for new functionality or bug fixes
+- Update documentation in `docs/` folder if needed
+- Follow the ["fork-and-pull" Git workflow](https://github.com/susam/gitpr)
+- At least one core contributor must review and approve changes
+- Run linters and tests before submitting
+
```

Comment:
Also, we want everyone to rebase against main and wait for passing tests before a PR is merged. One reason we do this is because tests run in the latest dev container built against the `main` branch. This means that if your PR updates our `images/Dockerfile`, the tests that get run for the PR aren't going to be run by GitHub actions in the image you're defining.